tests/tasks/dbt/test_dbt.py

```diff
@@ -0,0 +1,81 @@
+import os
+import pathlib
+import sys
+
+import pytest
+
+from prefect import Flow
+from prefect.tasks.dbt import DbtShellTask
+
+
+pytestmark = pytest.mark.skipif(
+    sys.platform == "win32", reason="DbtShellTask currently not supported on Windows"
+)
+
+
+@pytest.fixture()
+def test_path():
+    return os.path.dirname(os.path.realpath(__file__))
+
+
+def test_shell_creates_profiles_yml_file(test_path):
+    with Flow(name="test") as f:
+        task = DbtShellTask(
+            profile_name="default",
+            environment="test",
+            dbt_kwargs={
+                "type": "snowflake",
+                "threads": 1,
+                "account": "JH72176.us-east-1",
+                "user": "jane@company.com",
+                "role": "analyst",
+                "database": "staging",
+                "warehouse": "data_science",
+                "schema": "analysis",
+                "private_key_path": "/src/private_key.p8",
+                "private_key_passphrase": "password123",
+            },
+            overwrite_profiles=True,
+            profiles_dir=test_path,
+        )(command="ls")
+
+    out = f.run()
+    profiles_path = os.path.join(test_path, "profiles.yml")
+    assert out.is_successful()
+    assert os.path.exists(profiles_path)
+    os.remove(profiles_path)
+
+
+def test_shell_uses_dbt_envar(test_path):
+    dbt_project_path = os.path.join(test_path, "dbt_project")
+    os.environ["DBT_PROFILES_DIR"] = dbt_project_path
```

Comment:
We should use [monkeypatch] to patch the os environ like https://github.com/PrefectHQ/prefect/blob/7eb7a630be8b89ba569f17571c2b6bb090db82a7/tests/test_configuration.py#L59